gamla/currying.py

```diff
@@ -19,13 +17,10 @@ def _curry_helper(
 
         @functools.wraps(f)
         async def curry_inner_async(*args, **kwargs):
-            return await f(
-                *(args_so_far + args), **(toolz.merge(kwargs_so_far, kwargs))
-            )
+            return await f(*(args_so_far + args), {**kwargs_so_far, **kwargs})
 
         return curry_inner_async
 
-    @functools.wraps(f)
```

Comment:
make sure this works, e.g. run wcm on some trivial utterance

Reply:
I think this might not work. the CG has to know the function's signature so even if it works on bots we have today it can cause bugs later on.

Reply:
1. how do i run a bot on a different version of gamla without releasing a new one?
2. @nitzo so you're saying i shouldn't do it? what bugs can it cause in the future?

Reply:
1. install `gamla` from your local path (pip install -e <gamla_dir> ). You might need to uninstall it before.
2. If the CG gets a function with **kwargs it will throw an exception since it can't know what the function's signature is... Is the performance boost so big ? And if so we are wrapping a couple of times, why not remove all ?